packages/ckeditor5-remove-format/src/removeformatcommand.ts

```diff
@@ -52,7 +54,18 @@ export default class RemoveFormatCommand extends Command {
 					const itemRange = writer.createRangeOn( item );
 
 					for ( const attributeName of this._getFormattingAttributes( item, schema ) ) {
-						writer.removeAttribute( attributeName, itemRange );
+						if ( isGHSAttributeName( attributeName ) ) {
+							const oldAttribute = item.getAttribute( attributeName ) as GHSViewAttributes;
+							const newAttribute: GHSViewAttributes = {
+								...oldAttribute,
+								classes: [],
+								styles: {}
```

Comment:
I'm not sure if a direct modification of this model element attribute is a safe approach. What if we modify the internal format of how we store that data? Maybe we could use an approach similar to the style feature:

https://github.com/ckeditor/ckeditor5/blob/08270aafcc6e7e4fb4f4a4d2ebbc7451f031d303/packages/ckeditor5-style/src/stylecommand.ts#L201-L209

https://github.com/ckeditor/ckeditor5/blob/08270aafcc6e7e4fb4f4a4d2ebbc7451f031d303/packages/ckeditor5-html-support/src/generalhtmlsupport.ts#L131

Reply:
@niegowski I checked these methods and I see few problems:

1. These methods assume that the element contains only one GHS attribute, but there are cases when there are multiple formatting GHS on a single element. The example is list item [1]
2. `getGhsAttributeNameForElement` is used in these methods, so if we map model element to view element, it turns out that for list items it's not working. For example, in the [1] case these paragraph elements are resolved later to `htmlSpanAttribute` GHS attribute name, so we can't remove list item styling despite it's visible.
3. I'm not sure if we are able to remove all classes or elements according to logic of `modifyGhsAttribute` function (that is used in  `removeModelHtmlClass`) and this particular `if`: https://github.com/ckeditor/ckeditor5/blob/ck/13983/packages/ckeditor5-html-support/src/utils.ts#L180-L182

[1]
![obraz](https://github.com/user-attachments/assets/077f22cf-9c3f-4da3-bfda-79f974e72e04)